### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by woollybah** [bmx-ng/bmx-site#21]
There are still big gaps in the documentation.

I personally find writing docs hard work (although once I get into the zone, it's not so bad), which is why there are so many big gaps still.

Contributions are always welcome. Contributions don't necessarily need to be complete works.. even a basic draft for some particular subject is a good start, that could be padded out at some later time.

If you'd like to volunteer to write something, let us know which one you'd like to have a go at, and we'll assign it to you.

This is how things currently look..

### Setup
- [x] Getting Started
- [x] Win32
- [ ] Linux
- [ ] macOS
- [ ] Android
- [ ] iOS
- [ ] Raspberry Pi
- [ ] NX (homebrew)
- [ ] Custom Settings
### Tools
- [x] MaxIDE
- [ ] bmk (probably just needs updating?)
- [ ] bcc

### About
- [ ] About

**Comment by GWRon** [comment]
I think at least the newly added platforms (NX, raspi ...) need work done by you as you know what to setup there ... 
Also some stuff requires hardware to check it out (do not own a raspi, ios, nx ...)


Can try to write up something for "bcc" and update "bmk" as first step this evening.

**Comment by Hezkore** [comment]
I'd really like some setup info for Linux.
I've been doing a lot with BmxNG lately and 90% of runs on Linux, and I've had to use Wine to run it all.

**Comment by GWRon** [comment]
Can you explain what isn't working?

Error messages etc.

**Comment by Hezkore** [comment]
I'll have a look when I can, it was a long time since I last tried.
I remember Bah curl being a big issue in the past though.

**Comment by GWRon** [comment]
Bah.curl is a "3rd party" module... so something not so thoroughfully tested. Issues with it could be created in the corresponding repo.

Issues with getting ng to work on a platform could of course be avoided if a documentation listed required steps.

**Comment by Hezkore** [comment]
Not a whole lot happens when I try to compile an application.

![2019-08-09_16-44-25](https://user-images.githubusercontent.com/1727541/62787432-fb100c00-bac4-11e9-84e7-99c91077db10.gif)

**Comment by GWRon** [comment]
So
./bmk -v
does not expose the version?

Does
gdb -r bmk
Expose any crash?

**Comment by Hezkore** [comment]
gdb isn't even something I have installed.
But ./bmk -v does display version info.

I get a bit further with VSCode and my extension.
But it gets stuck here:
![Code_2019-08-09_17-08-08](https://user-images.githubusercontent.com/1727541/62789009-55f73280-bac8-11e9-81ff-c6f71f197161.png)

**Comment by GWRon** [comment]
No gdb?
So "gcc -v " does not print something too? If so.. install "build-essentials".

**Comment by GWRon** [comment]
Did you follow the release instructions for linux: https://github.com/bmx-ng/bmx-ng/releases

**Comment by Hezkore** [comment]
I did not follow those instructions. I didn't even know they existed!
(see, why isn't this on the site? hehe)

Installed and `gcc -v` does print a bunch of stuff.
But `gdb` is not a command.

I'll try to compile some stuff now.

**Comment by GWRon** [comment]
Yes I am with you about the missing instructions which are available in the releases-descriptions.

**Comment by Hezkore** [comment]
Okay so it seems to compile now if I manually throw the command at bmk.
But it's still stuck in VSCode, but I'm guessing that's a problem with the Remote SSH extension if anything.
And I believe I always had gcc installed (cause I've compiled a bunch of stuff in the past on this machine).
But I guess something else was missing, or outdated or... corrupt, not sure.

**Comment by GWRon** [comment]
The remote ssh thing should not vreate problems. maybe it borks up the working directory or so... bcc and bmk are a bit strict on how to call them (no symlinks please).

So get it to work with MaxIDE first ...and then try to get it to work in VS.

**Comment by Hezkore** [comment]
MaxIDE is sadly only graphical, and my Debian machine has no screen or even desktop.
It's Terminal Debian, so I can only do text based things.
Calling bmk myself in a "normal" SSH client I can compile my bmx code, so I guess that's fine. :)
But with the awesome power of VSCode I should be able to SSH in an do everything I normally can, but it just hangs up at the first step.
I'll have to investigate this.

Anyways, thanks for the help!
And that info really needs to be on the site (along with nightly builds!)

**Comment by GWRon** [comment]
Do you see how (command line) vs calls the binary?

Nightly builds require a build server...especially the Mac OS builds cannot be cross compiled while Windows and Linux builds could be built on a simple Linux server.

**Comment by Hezkore** [comment]
I did some debugging and found the cause to be the system PATH not being set correctly.
I can now compile Bmx code in VSCode via the Windows Subsystem for Linux, or by using a SSH connection to my Linux server computer. :)

Here's some more info: https://www.syntaxbomb.com/index.php/topic,5824.msg28540.html#msg28540
